### DISCUSSION TITLE
Uploading local image from blob

### DISCUSSION THREAD

**Discussion by YuriiTelishevsyi** [sibiraj-s/ngx-editor#489] [Q&A] [answered]
Hello. I want to create a custom upload image, not by the URL from the internet, but using localhost.
I want to make it with url like this 'blob:http://localhost:4200/6b3e6e2c-eccc-44f3-8118-6689ec7ed443' (url), but for now I can't. tried to make it with 
			`this.editor.commands.insertImage(url);`
but it does nothing. I have made it with a custom menu template and trying to use functional from insertImage.
can you please tell me what am I doing wrong? Or if it is even possible

**Comment by YuriiTelishevsyi** [comment]
![image](https://github.com/sibiraj-s/ngx-editor/assets/86302632/d52063a6-ff03-473c-99cc-84b3dcffd650)

**Reply by YuriiTelishevsyi** [reply]
even tried customly hardcode real image url didn't work
![image](https://github.com/sibiraj-s/ngx-editor/assets/86302632/8ff37d73-6531-4929-b805-ee1ee91dea23)

**Reply by sibiraj-s** [reply, marked as answer]
> tried to make it with
> this.editor.commands.insertImage(url);
> but it does nothing

Its a chain. You must call exec at end of the commands `this.editor.commands.insertImage(url).exec()` to apply the transaction

**Reply by sibiraj-s** [reply]
see https://sibiraj-s.github.io/ngx-editor/en/commands/

**Reply by YuriiTelishevsyi** [reply]
worked. thanks

[a comment by an automated account is not shown]
